src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.RoleDao;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Role> query = criteriaBuilder.createQuery(Role.class);
+            Role.RoleName role = null;

```

Comment:
You can simplify  this method  if use Query instead CriteriaBuilder

Reply:
Nope I like using criteria API as well. Also, number of code lines won't really change if I'll use HQL.